src/pages/_app.jsx

```diff
@@ -121,6 +106,47 @@ const WrappedApp = ({ Component, pageProps, req }) => {
   );
 };
 
+WrappedApp.getInitialProps = async ({ Component, ctx }) => {
+  const {
+    store, req, query, res,
+  } = ctx;
+
+  const pageProps = Component.getInitialProps
+    ? await Component.getInitialProps(ctx)
+    : {};
+
+  const promises = [];
+
+  if (req) {
+    const { default: getAuthenticationInfo } = require('../utils/ssr/getAuthenticationInfo');
+    promises.push(getAuthenticationInfo);
+
+    if (query?.experimentId) {
+      const { default: getExperimentInfo } = require('../utils/ssr/getExperimentInfo');
+      promises.push(getExperimentInfo);
+    }
+  }
+
+  try {
+    let results = await Promise.all(promises.map((f) => f(ctx, store)));
+    results = _.merge(...results);
+
+    return { pageProps: { ...pageProps, ...results } };
+  } catch (e) {
+    if (e instanceof CustomError) {
+      if (res && e.payload.status) {
+        res.statusCode = e.payload.status;
+      } else {
+        res.statusCode = 500;
+      }
+
+      return { pageProps: { ...pageProps, httpError: e.payload.status || true } };
+    }
+
+    throw e;
+  }
+};
+
```

Comment:
The actual server-side rendering logic. Feel free to ask any questions you may have.

Reply:
can you put a comment in the code that this is the SSR logic, so it is more obvious?

Reply:
it's just a suggestion